src/firefly/js/visualize/saga/ChartsSync.js

```diff
@@ -34,9 +34,21 @@ export function* syncCharts() {
             case TablesCntlr.TABLE_SORT:
             case TablesCntlr.TABLE_NEW_LOADED:
                 const {tbl_id} = action.payload;
-                tableStatsState = flux.getState()[TableStatsCntlr.TBLSTATS_DATA_KEY];
-                if (has(tableStatsState, tbl_id)) {
-                    TableStatsCntlr.dispatchLoadTblStats(request);
+
+                // table statistics and histogram data do not change on table sort
+                if (action.type !== TablesCntlr.TABLE_SORT) {
+                    tableStatsState = flux.getState()[TableStatsCntlr.TBLSTATS_DATA_KEY];
+                    if (has(tableStatsState, tbl_id)) {
```

Comment:
just a comment on lodash's 'has'.  it will return true as long as tbl_id property is defined.  its value can be anything, including null, undefined, or ''.  in this case, i think a 'get' would be more precise.